### PR TITLE
git ignore dist files

### PR DESCRIPTION
**Pull request by unDemian**
**Built on top of https://github.com/Automattic/happychat-client/pull/176**

## Summary
Remove dist files for all targets from repo and add them to gitignore. These can be re-build at whenever needed.

## Testing
Standalone chat should work as before, `npm start`. Building for all targets `npm run targets` should create proper dist files but not add them to the repo.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
 node_modules/
+*.map
+
+# Project
 tokens
-targets/standalone/public/tokens
-targets/standalone/public/happychat.css
-targets/wordpress/assets/happychat.css
 stats.json
+*.local.json
```
Comment (lamosty) [inline .gitignore]: 💯

**Comment by nosolosw** [comment]
What would be the purpose of the `dist` target if we remove its assets? Do you think we should remove it?

Thinking out loud: the `dist` target was only useful for maintaining the assets we were going to distribute through WordPress.com CDN at `https://widgets.wp.com/happychat/happychat.css`. So:

- I lean towards thinking it's still useful to perhaps keep the CSS in this target? If it's within the codebase is very easy to know when it needs to be updated in WordPress.com. `npm run targets:dist:css` or download `https://widgets.wp.com/happychat/happychat.css` to `targets/dist/` and right away you know it: if happychat.css was changed, it needs to be updated in wpcom. If we lose this, computing the differences becomes messy and I'm not sure how to do it. What do you think we should do if we don't have this?

- The `dist` target also contained the production JS. My thinking was that I'd have the time to create a minimal core JS that will download other bundles on demand ([see](https://github.com/Automattic/happychat-client/issues/10)) - and that core will be distributed through our CDN. I didn't had the time to do this, so it may be as well removed from here. If we do this, though, we should create production and development versions of the JS for each target (right now all but `dist` use the development versions).

**Comment by unDemian** [comment]
> What would be the purpose of the dist target if we remove its assets? Do you think we should remove it?

In a future PR this will be renamed to browser and I think we should keep it around as the plan for Happychat is to have an embeddable client that works with any site.

> I lean towards thinking it's still useful to perhaps keep the CSS in this target? If it's within the codebase is very easy to know when it needs to be updated in WordPress.com. npm run targets:dist:css or download https://widgets.wp.com/happychat/happychat.css to targets/dist/ and right away you know it: if happychat.css was changed, it needs to be updated in wpcom. If we lose this, computing the differences becomes messy and I'm not sure how to do it. What do you think we should do if we don't have this?

We should lose that flow, all we have to do is build the css which will be gitignored and compare that file with the one in production, however I think we should consider a more reliable versioning strategy, maybe something like https://www.npmjs.com/package/webpack-auto-inject-version

> The dist target also contained the production JS. My thinking was that I'd have the time to create a minimal core JS that will download other bundles on demand (see) - and that core will be distributed through our CDN. I didn't had the time to do this, so it may be as well removed from here. If we do this, though, we should create production and development versions of the JS for each target (right now all but dist use the development versions).

In a future PR, npm commands are added for all targets for both development and production, and yes this should remain the "core" target.

**Comment by unDemian** [comment]
Thanks for taking the time to review and discuss this 🙇

**Pull request merged**